followthemoney/multi_part_proxy.py

```diff
@@ -0,0 +1,92 @@
+from followthemoney.exc import InvalidData
+from followthemoney.proxy import EntityProxy
+
+
+class MultiPartProxy(EntityProxy):
+    def __init__(self, id=None, proxies=None, anti_proxies=None):
+        self.id = id
+        self.proxies = set(proxies or [])
+        self.anti_proxies = set(anti_proxies or [])
+        self._golden_proxy = None
+        self._build()
+
+    def _build(self):
```

Comment:
Just an alternative idea: what if we added a `clear()` method to the EntityProxy that flushes it's props and then just used the actual `self` for the storage, while keeping `add()`, `set()`, `pop()` and `remove()` overwritten? So `_build` would `clear`, then re-fill `self`....

Reply:
That's what I had wanted to do but there are multiple issues that make the implementation pretty dirt, most annoyingly:

1. the clear becomes a very specific to the multi_part_proxy in that it should probably keep context and existing IDs
2. many methods, including merge, break because `add` is not implemented. 

This was one of the motivators behind me questioning whether this object should be inheriting from `EntityProxy` at all as opposed to being a container.